integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -88,21 +120,36 @@ fn test_resharding_v3() {
     }
     let (genesis, _) = genesis_builder.build();
 
-    let TestLoopEnv { mut test_loop, datas: node_datas, tempdir } = builder
+    let mut builder = builder
         .genesis(genesis)
         .epoch_config_store(epoch_config_store)
         .clients(clients)
-        .track_all_shards()
-        .build();
+        .track_all_shards();
+    if !chunk_ranges_to_drop.is_empty() {
+        builder = builder
+            .drop_protocol_upgrade_chunks(base_protocol_version + 1, chunk_ranges_to_drop.clone());
+    }
+    let TestLoopEnv { mut test_loop, datas: node_datas, tempdir } = builder.build();
 
     let client_handle = node_datas[0].client_sender.actor_handle();
+    let latest_block_height = AtomicU64::new(0);
     let success_condition = |test_loop_data: &mut TestLoopData| -> bool {
         let client = &test_loop_data.get(&client_handle).client;
         let tip = client.chain.head().unwrap();
 
         // Check that all chunks are included.
         let block_header = client.chain.get_block_header(&tip.last_block_hash).unwrap();
-        assert!(block_header.chunk_mask().iter().all(|chunk_bit| *chunk_bit));
+        if latest_block_height.load(Ordering::SeqCst) < tip.height {
+            if latest_block_height.load(Ordering::SeqCst) == 0 {
+                println!("State before resharding:");
+                print_and_assert_shard_accounts(client);
+            }
+            latest_block_height.store(tip.height, Ordering::SeqCst);
```

Comment:
nit: I would feel a tiny bit better having the `latest_block_height` in a mutex and locked for the entire check.

Reply:
I'll just use `std::cell::Cell`. I only need to mutate the counter in the function.